### PR TITLE
Post Ruma upgrade cleanup

### PR DESCRIPTION
**Pull request by jplatte**
Lots of small cleanup changes and a few select edge case behavior changes. Mostly based on what I noticed when upgrading Ruma.

### CODE REVIEW

File: crates/matrix-sdk-crypto/src/backups/mod.rs [hunk 1]
```diff
@@ -463,57 +463,4 @@ mod test {
...
-        let machine =
-            OlmMachine::with_default_store(alice_id(), alice_device_id(), tmpdir.as_ref(), None)
-                .await?;
-
-        backup_flow(machine).await
-    }
-
-    #[async_test]
-    #[cfg(feature = "sled_cryptostore")]
-    async fn recovery_key_storing() -> Result<(), OlmError> {
```
Comment (poljar) [inline crates/matrix-sdk-crypto/src/backups/mod.rs:482]: #558 moves this test into the integration tests instead of just removing it. The other `sled_cryptostore` dead code is also removed in #558.

Comment (jplatte) [reply crates/matrix-sdk-crypto/src/backups/mod.rs:482]: Okay, I'll remove this commit then.

**Review by poljar — APPROVED** [review]
Looks good, the usize indeed isn't portable and lets remove the commented code or at lest add a reason why it should stay.

File: crates/matrix-sdk-sled/src/encode_key.rs [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+}
+
+/* impl EncodeKey for RoomAccountDataEventType {
+    fn encode(&self) -> Vec<u8> {
+        self.to_string().encode()
+    }
+} */
+
+pub fn encode_key_with_usize<A: AsRef<str>>(s: A, i: usize) -> Vec<u8> {
+    // Not portable across architectures - is this right?
```
[the remark this answers is not in the document]
Comment (jplatte) [reply crates/matrix-sdk-sled/src/encode_key.rs:155]: Wdym by "depending on if this needs to be decoded"?

[the remark this answers is not in the document]
Comment (poljar) [reply crates/matrix-sdk-sled/src/encode_key.rs:155]: This encodes the tuple (A, usize) into a `Vec<u8>`. If we need to perform the reverse operation, decode a `Vec<u8>` back into `(A, usize)`, we might want to consider using an `u32`. This means we fail at the encoding step on 64 bit architectures instead of failing at the decoding step on 32 bit architectures.

That being said, I have not seen that this needs to be decoded, though I have not thoroughly checked.

[the remark this answers is not in the document]
Comment (jplatte) [reply crates/matrix-sdk-sled/src/encode_key.rs]: Left a FIXME comment and opened #561 because this isn't as straight-forward as the refactorings in this PR.
